packages/ra-core/src/controller/details/useCreateController.ts

```diff
@@ -118,6 +115,11 @@ export const useCreateController = <
     const recordToUse = getRecord(location, record);
     const version = useVersion();
 
+    const { resource } = useResource(props);
+    const { hasCreate, hasEdit, hasShow, hasList } = useResourceConfig(
```

Comment:
I don't think the controller should grab that data. The controller should only grab data necessary to build a context. The resource configuration data is already available in Redux. Individual components that need the resourceConfig can simply use it from the redux store (which is BC because the data already exists in the store).